### ISSUE TITLE
TWE: Create Template Issue: Research Plans and Goals Chart Template for Each Participant Type on Wiki

### ISSUE THREAD

**Issue by pandanista** [hackforla/internship#432]
### Overview
We need to create a template issue for research plans and goals chart so that we have consistent, transparent, defined and self-supporting structure for people creating them.

### Action Items
- [x] Add to the Project Board
- [x] Add the milestone: Project Management
- [x] Add a feature label specific to what the template is related to, if available.  If not clear what you should use, ask product.
- [x] Any place you see `[replace...]` replace both the text and the brackets with what it is asking for
- [x] Create a draft of new template by copying the content block below to a comment on this issue, which will be known as our Draft
- [x] In your draft, Fill out the top part (called front matter) with the info it needs.  See the Resources section below for definition and example.
- [x] Define any Dependencies (or remove the dependency section)
- [x] Customize the Overview section
- [x] Fill out the Action Items section (in your draft) by defining the steps that are required to be done, keeping the following in mind:
   - [x] The person working on this issue might be new to the org, so the issue should be able to be worked on without any prior knowledge
   - [x] If the person needs to add links, or put documents in specific locations, the place where they can do that action must be linked in the resources section
      - [x] the links provided by the issue creator should be in the same order as the Action Items 
- [x] If documents or folders (sheets, docs, presentations, dashboard, etc.) will be created, find the document template that should be used by checking the WIKI Templates by role pages.  
  - [x] if there is a template
     - [x] Add a link to the template into the ` ### Resources`  section of your draft
  - [ ] if there is no template yet
    - [ ] put a placeholder into the` ### Resources`  section of the draft `[TWE: Template: [replace with name of type]]`
- [ ] Get sign-off by Product on the draft

#### If this is for a template draft that needs testing at a later stage, please use the following handoff steps:
- [x] Add `feature: template creation (step 1)` to this issue
- [x] Once the template draft is finished, assign a team member or yourself to create testing issues 
- [ ] Once the testings issues are completed, get sign-off by Product on the template draft
- [ ] Create or update the template issue on the repo 
- [ ] Add the link to the new issue template once created to resource 2.01.  Find by clicking on the **Get started** button to the right of the new template listed on [new issue page](https://github.com/hackforla/internship/issues/new/choose)
- [ ] Update the appropriate place in the wiki for your template (see resources section)

#### Resources for creating this issue
- 1.01 Content to copy for draft
```
Draft of template issue for `[Replace NAME OF ISSUE TEMPLATE]`
---
**name:** TWE: Template Issue: [Replace ROLE]: [Replace ACTION VERB][Replace DIRECT OBJECT]
**about:** This issue helps you structure your template issue to be consistent with our other template issues in [Replace DIRECT OBJECT]
** title:** TWE: [Replace ROLE]: [Replace DIRECT OBJECT]: [Replace ACTION VERB]
**labels:** 'feature: board / google drive / tools, feature: missing, milestone: missing, role: missing, size: missing', 'Participant Type: missing', 'complexity: missing'
**assignees:**

---
#### Instructions for creating this issue
The person who creates this issue should use these links to add links to the resources section and customize this issue
- Customize the issue
    - [ ] Copy and paste entire template text in this issue into text editor (ie., "TextEdit" on _Mac_, "Wordpad" on _PC_)
    - Use Find/Replace (**Mac:** ⌘+F or **PC**: Ctrl+F) to search for and replace the keyword strings (including text and the brackets) below with the appropriate values 
    - [ ] Search for `[Replace`
    - [ ] Replace with whatever the keyword or information is asking for.
- Customize the issue settings (located in right sidebar)
    - [ ] Under Labels, add labels for any labels identified as missing
    - [ ] Remove the corresponding missing labels
    - [ ] Under Projects, choose gear, choose "Repository," and choose TWE: Project Board.
    - [ ] Add the milestone: [Replace MILESTONE]
    - [ ] Add a feature label specific to what the template is related to, if available. If not clear what you should use, ask product.
- Customize dependencies 
    - [ ] If no dependencies, delete dependency section
    - [ ] If dependencies, add the dependencies (any issue or reason that would prevent the action items from being completed)
- Remove this section.
    - [ ] Delete this section when all the checkboxes above are complete (i.e., the issue assignee will never see this part)

### Dependencies

### Overview
We need to [Replace ACTION VERB] [Replace DIRECT OBJECT] for [Replace REASON].

### Action Items
A STEP BY STEP LIST OF ALL THE TASK ITEMS THAT YOU CAN THINK OF NOW EXAMPLES INCLUDE: Research, reporting, coding, etc.

### Resources
#### Resources for creating this issue
- 1.01
- 1.02

#### Resources gathered during the completion of this issue
- 2.01
- 2.02
```

- 1.02 Front Matter Resources
   <details>
   <summary>Front Matter Fields and Meaning</summary>

   - name: (name that appear on the new issue tab)
   - about: (is the description that appears next to the issue template on the new issue tab)
   - title: (the title that appears when the issue is created.  User then customizes this title for their issue)
       - Use the naming convention: TWE: Role: Direct object: Action verb
       - If more than one word should go into the direct object spot, use multiple square brackets with the keyword (see example below)
   - labels: (choose the labels that will be the same each time, and choose missing labels for the ones that depend on content)
   - assignees:  
   </details>
   <details>
   <summary>Front Matter Field Example</summary>

   - name: 'TWE: Template Issue: Research: Create Roadmap'
   - about:  This issue helps you structure your template issue to be consistent with our other template issues in roadmap
   - title: 'TWE: Research: IS[Replace YY]: RP[Replace 000]: [Replace TYPE OF PARTICIPANT] [Replace TYPE OF RESEARCH METHOD]: Roadmap : Create'
   - labels: 'feature: research', 'feature: roadmap', 'milestone: missing', 'role: UI/UX research', 'Participant Type: missing', 'Research 000: missing', 'complexity: missing'
   - 'size: 0.50pt'
   - assignees: ''
   </details>

- 1.03 Example Action Items for Finding Wiki Page and Template
   <details>
   <summary>Example Action Items</summary>

   ```
   ### Action Items
   - Customize wiki page
        - [ ] Go to link 1.01 Internship Wiki, Join the team page
        - [ ] Find wiki page for this issue by looking at role pages.
        - [ ] Copy the link for the wiki page.
        - [ ] Add the link to Resource 2.01
           - [ ] Add the name of the link in square brackets
           - [ ] Add the URL in parenthesis without a space between the square bracket and parenthesis
   - Customize document template.
       - [ ] Go to link 1.02 Design Process Overview
       - [ ] Find document template that should be used in this issue.
       - [ ] Copy the link for the template.
       - [ ] Add the link to Resource 2.02
           - [ ] Add the name of the link in square brackets
           - [ ] Add the URL in parenthesis without a space between the square bracket and parenthesis
   ```

   </details>

   <details>
   <summary>Example Resource Lists</summary>

   ```
   ### Resources for creating this issue
   - 1.01. [Internship Wiki, Join the team page](https://github.com/hackforla/internship/wiki/Joining-the-Team)
   - 1.02. [Design Process Overview](https://github.com/hackforla/internship/wiki/Design-Process-Overview)

   ### Resource links gathered during the making of this issue
   - 2.01. [Wiki page for [Replace TYPE OF ROLE]]
   - 2.02. [TWE: Template: [Replace NAME OF TEMPLATE]]
   ```

   </details>

- 1.04 WIKI Templates by role pages
   - [TWE: WIKI: UX Research Guides Templates](https://github.com/hackforla/internship/wiki/UX-Research-Guides-Templates) 
   - [Design](https://github.com/hackforla/internship/wiki/Design-Process-Overview)
   - [Product](https://github.com/hackforla/internship/wiki/Product)
   - [Business Analyst / Data Science]
   - [Software Engineer]
   - [Marketing]
   - default document templates, in case you don't have one yet, can be found on the [Design Process Overview page](https://github.com/hackforla/internship/wiki/Design-Process-Overview)

#### Resources gathered during the completion of this issue
- 2.01 [TWE: [Replace ROLE]: [Replace DIRECT OBJECT]: [Replace ACTION VERB]]([Replace LINK])

**Comment by pandanista** [comment]
#### Instructions for creating this issue
The person who creates this issue should use links in the section "Resources for creating this issue" (i.e., 1.01, 1.02, etc.) to customize this issue.
- Customize the issue
    - [ ] Open the "Research Overview" wiki page (Resource 1.01)
    - [ ] Choose the appropriate research roadmap or research plan folder (ie., RP###) on the wiki page 
    - Use the information on the wiki page and in research roadmap or research plan to customize this issue
    - [ ] Copy and paste entire template text in this issue into text editor (ie., "TextEdit" on _Mac_, "Wordpad" on _PC_)
    - Use Search/Replace (**Mac:** ⌘+F or **PC**: Ctrl+F) to search for and replace the keyword strings (including text and the brackets) below with the appropriate values 
        - Year 
            - [ ] Search `[Replace YY]` and replace all with research year (ie., "22" for 2022)
        - Research Plan Number
             - [ ] Search `[Replace 000]` and replace all with the research plan number (ie., "001" for RP001)
        - Type of Participant
            - [ ] Search `[Replace TYPE OF PARTICIPANT]` and replace all with the participant type (ex., "Intern")  
        - Type of Interview
            - [ ] Search `[Replace TYPE OF INTERVIEW]` and replace all with the type of interview (ex., "Exit")
    - [ ] Copy and paste the edited template from text editor back into this Github Issue, replacing all original text
    - [ ] Re-check all the checkboxes above after copying and pasting the edited template
    - [ ] Close the text editor
    - Update the issue title
        - [ ] Choose "Edit" in the title section
        - [ ] Copy and paste 
        ```
        TWE: IS[Replace YY]: RP[Replace 000]: [Replace TYPE OF PARTICIPANT] [Replace TYPE OF INTERVIEW] Interview Roll Call and Session Table: Create` into the issue title
        ```
        - [ ] Choose "Save"
- Customize the issue settings (located in right sidebar)
    - [ ] Under Assignees, assign a Researcher if it has been determined. Otherwise, leave it blank.
    - [ ] Under Labels,  choose gear  ![Screenshot 2023-01-11 at 10 50 11 AM](https://user-images.githubusercontent.com/57029070/211904252-46521281-b8f8-495d-b31f-d06e425f0096.png), and apply the label `Research: RP[Replace 000]`
    - [ ] Under Labels, choose gear  ![Screenshot 2023-01-11 at 10 50 11 AM](https://user-images.githubusercontent.com/57029070/211904252-46521281-b8f8-495d-b31f-d06e425f0096.png), and apply the label `Participant Type: [Replace TYPE OF PARTICIPANT]`
    - [ ] Remove the `Research 000: missing` and `Participant Type: missing` labels
    - [ ] Under Projects, choose gear ![Screenshot 2023-01-11 at 10 50 11 AM](https://user-images.githubusercontent.com/57029070/211904252-46521281-b8f8-495d-b31f-d06e425f0096.png), choose `Repository`, and choose `Program Management`
    - [ ] Add the milestone: choose gear ![Screenshot 2023-01-11 at 10 50 11 AM](https://user-images.githubusercontent.com/57029070/211904252-46521281-b8f8-495d-b31f-d06e425f0096.png), and choose `Research Plan Execution`
    - [ ] Remove the label `milestone: missing`
- Remove this section
    - [ ] After all the checkboxes above are complete and the issue is ready to be worked on, delete all text below the "Instructions for Creating this Issue" (top heading) and above "Overview" (heading below) -- i.e., the issue assignee will never see this part

**Comment by pandanista** [comment]
@priyatalwar suggested this template to have epic label added. We would like confirmation on that.

**Comment by pandanista** [comment]
Adding "ready for product" label and moving this template issue to the review column as it is currently being tested by Product Manager (see issue #445).
 
This template has passed the first round of internal testing (issue #374).

**Comment by sunannie27** [comment]
@priyatalwar 
Please provide update
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures or links* (if necessary): "Add any pictures or links that will help illustrate what you are working on."

* remember to add links to the top of the issue if they are going to be needed again.

**Comment by pandanista** [comment]
I've made drastic changes to this template (the wording, the structure, and the instructions) based on PM's feedback on 10/10. 

All the changes were incorporated in testing issue #445 while keeping the part PM has tested and approved up-to-date (see all the checkmarks and clickable URLs). Two options for the wording regarding the parenthesis are included in testing issue #445 as well. The request for PM to continue the testing has made in #445.

**Comment by pandanista** [comment]
- Testing issues: #445 and #374
- Incorporated feedback from testers
- Updated the create a research plan template link and TWE: project board link 

Next step:
- Ready to turn this into a template on a repo.

**Comment by pandanista** [comment]
Closing this issue as [the create research plans and goals chart template](https://github.com/hackforla/internship/issues/new?assignees=&labels=complexity%3A+medium%2C+feature%3A+overview%2C+feature%3A+research+plan%2C+milestone%3A+missing%2C+p-feature%3A+wiki%2C+Participant+Type%3A+missing%2C+Research+000%3A+missing%2C+role%3A+UI%2FUX+research%2C+size%3A+1pt&projects=&template=research--create-a-research-plans-and-goals-chart-on-wiki.md&title=TWE%3A+IS%5BReplace+YY%5D%3A+%5BReplace+TYPE+OF+PARTICIPANT%5D+Research+Plans+and+Goals+Chart%3A+Create) is deployed on the repo.
